src/components/transport_manager/src/usb/libusb/usb_handler.cc

```diff
@@ -89,8 +89,27 @@ UsbHandler::UsbHandler()
 }
 
 UsbHandler::~UsbHandler() {
-  shutdown_requested_ = true;
-  SDL_LOG_INFO("UsbHandler thread finished");
+  SDL_LOG_AUTO_TRACE();
+
+  // Notify Thread to stop on next iteration...
+  RequestStopThread();
+
+  // ... and unregister hotplug callbacks in order to wake
+  // libusb_handle_events()
+  DeregisterHotplugCallbacks();
+
+  // Now it is safe to join the Thread and free up resources
+  JoinAndDeleteThread();
+  InvokeLibusbExit();
+}
+
+void UsbHandler::RequestStopThread() {
+  SDL_LOG_AUTO_TRACE();
+  shutdown_requested_.exchange(true);
```

Comment:
Oh no, here we go again :)) 
Why `exchange` instead of `store`?

Reply:
@YarikMamykin its a secret code to confuse enemies. Fixed in https://github.com/smartdevicelink/sdl_core/pull/3492/commits/51c490c263a9eebb67b0a4caa4a701b8fcd22787